### PR TITLE
New package: logisim-evolution-3.8.0

### PR DESCRIPTION
**Pull request by voidbert**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl

### CODE REVIEW

File: srcpkgs/logisim-evolution/template [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+revision=1
+build_style=fetch
+depends="openjdk17-jre"
+short_desc="Digital logic design tool and simulator"
+maintainer="voidbert <humbertogilgomes@protonmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/logisim-evolution/logisim-evolution"
+distfiles="
+ https://github.com/logisim-evolution/logisim-evolution/releases/download/v${version}/logisim-evolution-${version}-all.jar
+ https://raw.githubusercontent.com/logisim-evolution/logisim-evolution/v${version}/artwork/logisim-evolution-icon.svg"
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:13]: we should be building this ourselves, not using prebuilt artifacts

Comment (voidbert) [reply srcpkgs/logisim-evolution/template:13]: Thank you for the guidance! I am still an xbps packaging newbie. I'll address these issues and update the PR.

**Comment by voidbert** [comment]
@classabbyamp I have fixed all problems. Though, `xlint` is still complaining about my email (likely commit email), when I have entered my real email on the maintainer field. Is that OK? If not, what should I do? `git config --global user.email` didn't work.

**Comment by classabbyamp** [comment]
yes, it's talking about the email git is using to commit. it's still erroring because you added more commits instead of amending and/or squashing

File: srcpkgs/logisim-evolution/template [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	export JAVA_HOME=/usr/lib/jvm/openjdk17
+	gradle --no-daemon shadowJar
+}
+
+do_install() {
+	vmkdir usr/share/logisim-evolution
+	vmkdir usr/share/applications
+	vmkdir usr/bin
+
+	vcopy build/libs/logisim-evolution-${version}-all.jar usr/share/logisim-evolution
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:31]: use vinstall instead, and this should go in `usr/lib/$pkgname/` instead

File: srcpkgs/logisim-evolution/template [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Template file for 'logisim-evolution'
+pkgname=logisim-evolution
+version=3.8.0
+revision=3
+hostmakedepends="openjdk17 gradle sed tar"
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:5]: don't include sed here, it should already be there.

why is tar needed?

File: srcpkgs/logisim-evolution/template [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+do_install() {
+	vmkdir usr/share/logisim-evolution
+	vmkdir usr/share/applications
+	vmkdir usr/bin
+
+	vcopy build/libs/logisim-evolution-${version}-all.jar usr/share/logisim-evolution
+
+	sed -e "s/@VERSION@/${version}/g" ${FILESDIR}/logisim-evolution > \
+		${DESTDIR}/usr/bin/logisim-evolution
+	chmod 0755 ${DESTDIR}/usr/bin/logisim-evolution
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:35]: use `vbin $FILESDIR/logisim-evolution` then sed after

File: srcpkgs/logisim-evolution/template [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+	vcopy artwork/logisim-evolution-icon.svg usr/share/logisim-evolution
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:37]: use vinstall, and this should go in /usr/share/icons/hicolor/scalable/apps/ with a name that matches the Icon= line in the desktop file

File: srcpkgs/logisim-evolution/template [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	vcopy support/Flatpak/com.github.reds.LogisimEvolution.desktop usr/share/applications
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:38]: vinstall

File: srcpkgs/logisim-evolution/template [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+do_build() {
+	export JAVA_HOME=/usr/lib/jvm/openjdk17
+	gradle --no-daemon shadowJar
+}
+
+do_install() {
+	vmkdir usr/share/logisim-evolution
+	vmkdir usr/share/applications
+	vmkdir usr/bin
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:29]: should not be needed if using vinstall/vbin

File: srcpkgs/logisim-evolution/template [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+do_install() {
+	vinstall build/libs/logisim-evolution-${version}-all.jar 0644 usr/lib/${pkgname}
+
+	vbin ${FILESDIR}/logisim-evolution
+	sed -i -e "s/@VERSION@/${version}/g" -e "s/@PKGNAME@/${pkgname}/g" \
+		${DESTDIR}/usr/bin/logisim-evolution
+
+	vinstall artwork/logisim-evolution-icon.svg 0644 \
+		usr/share/icons/hicolor/apps com.github.reds.LogisimEvolution.svg
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:28]:
```suggestion
		usr/share/icons/hicolor/scalable/apps com.github.reds.LogisimEvolution.svg
```

**Review by classabbyamp — COMMENTED** [review]
also add the mimeinfo file:
```sh
vinstall support/Flatpak/com.github.reds.LogisimEvolution.desktop 0644 usr/share/mime/packages
```

**Comment by classabbyamp** [comment]
to summarise:
```patch
From 528ba136392986688f1bfb7c4327fcf9125e1035 Mon Sep 17 00:00:00 2001
From: classabbyamp <void@placeviolette.net>
Date: Tue, 17 Jan 2023 13:15:15 -0500
Subject: [PATCH] pr review


diff --git a/srcpkgs/logisim-evolution/files/logisim-evolution b/srcpkgs/logisim-evolution/files/logisim-evolution
index 1039e777fb..7fdaedabdd 100644
--- a/srcpkgs/logisim-evolution/files/logisim-evolution
+++ b/srcpkgs/logisim-evolution/files/logisim-evolution
@@ -1,6 +1,4 @@
 #!/bin/sh
 
 export JAVA_HOME=/usr/lib/jvm/openjdk17
-$JAVA_HOME/bin/java -jar \
-       /usr/lib/@PKGNAME@/logisim-evolution-@VERSION@-all.jar $@
-
+$JAVA_HOME/bin/java -jar /usr/lib/logisim-evolution/logisim-evolution.jar $@
diff --git a/srcpkgs/logisim-evolution/template b/srcpkgs/logisim-evolution/template
index b59cd3c716..b89cacbb82 100644
--- a/srcpkgs/logisim-evolution/template
+++ b/srcpkgs/logisim-evolution/template
@@ -18,14 +18,11 @@ do_build() {
 }
 
 do_install() {
-       vinstall build/libs/logisim-evolution-${version}-all.jar 0644 usr/lib/${pkgname}
+       vinstall build/libs/logisim-evolution-${version}-all.jar 0644 usr/lib/logisim-evolution logisim-evolution.jar
 
        vbin ${FILESDIR}/logisim-evolution
-       sed -i -e "s/@VERSION@/${version}/g" -e "s/@PKGNAME@/${pkgname}/g" \
-               ${DESTDIR}/usr/bin/logisim-evolution
 
-       vinstall artwork/logisim-evolution-icon.svg 0644 \
-               usr/share/icons/hicolor/apps com.github.reds.LogisimEvolution.svg
-       vinstall support/Flatpak/com.github.reds.LogisimEvolution.desktop 0644 \
-               usr/share/applications
+       vinstall artwork/logisim-evolution-icon.svg 0644 usr/share/icons/hicolor/scalable/apps com.github.reds.LogisimEvolution.svg
+       vinstall support/Flatpak/com.github.reds.LogisimEvolution.desktop 0644 usr/share/applications
+       vinstall support/Flatpak/com.github.reds.LogisimEvolution.desktop 0644 usr/share/mime/packages
 }
-- 
2.39.0
```

File: srcpkgs/logisim-evolution/template [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+	gradle --no-daemon shadowJar
+}
+
+do_install() {
+	vinstall build/libs/logisim-evolution-${version}-all.jar 0644 usr/lib/logisim-evolution logisim-evolution.jar
+
+	vbin ${FILESDIR}/logisim-evolution
+
+	vinstall artwork/logisim-evolution-icon.svg 0644 usr/share/icons/hicolor/scalable/apps com.github.reds.LogisimEvolution.svg
+	vinstall support/Flatpak/com.github.reds.LogisimEvolution.desktop 0644 usr/share/mime/packages
```
Comment (classabbyamp) [inline srcpkgs/logisim-evolution/template:26]:
```suggestion
	vinstall support/Flatpak/com.github.reds.LogisimEvolution.xml 0644 usr/share/mime/packages
```
my bad -- i typo'd

Comment (voidbert) [reply srcpkgs/logisim-evolution/template]: No problem. And thanks for your patience!

**Pull request merged**